proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,340 @@
+* CSS Logical Properties
+  * Support CSS logical property names, i.e., `paddingInlineStart` rather than only non-standard `paddingStart`.
+  * Support subtree-level logical properties, where setting the `dir` prop to `ltr` or `rtl` alters the way logical properties are resolved. Currently, React Native only supports global writing direction configuration for Android.
+* CSS Animations
+  * Support declarative keyframes and animations that can be optimized on the native side and avoid the need for `Animated`.
+  * See [animation](https://developer.mozilla.org/en-US/docs/Web/CSS/animation)).
+* CSS Transitions
+  * Support declarative transitions that can be optimized on the native side and avoid the need for `Animated`.
+  * Support `transitionProperty`, `transitionDuration`, `transitionTimingFunction`, and `transitionDelay` (see [transition](https://developer.mozilla.org/en-US/docs/Web/CSS/transition)).
+* `aspectRatio`
+  * Support string values, i.e., `'16 / 9'`, to align with CSS.
+* `backgroundImage`
+  * Support setting background images via `url()`.
+  * Optional: support CSS gradients.
+* `boxShadow`
+  * Add native support for CSS box shadows.
+  * This would replace the very buggy iOS-specific `shadow*` styles and Android `elevation` style, allowing developers to unify shadows across native and web platforms.
+* `filter`
+ * Support declarative filters per [filter](https://developer.mozilla.org/en-US/docs/Web/CSS/filter)
+* `fontVariant`
+  * Support space-separated string values to align with CSS.
+* `fontWeight`
+  * Support number values to align with React DOM / CSS.
+* `lineClamp`
+  * Support CSS clamping text to a set number of lines via style.
+  * This would be equivalent to using the `numberOfLines` prop on `<Text>` components.
+* `objectFit`
+  * Support CSS `objectFit` property for `<Image>`.
+  * This would be a subset (no "repeat" keyword) of functionality currently supported by the `resizeMode` prop and style for `<Image>`.
+  * `objectFit === 'contain'` => `resizeMode = 'contain'`
+  * `objectFit === 'cover'` => `resizeMode = 'cover'`
+  * `objectFit === 'fill'` => `resizeMode = 'stretch'`
+  * `objectFit === 'scale-down'` => `resizeMode = 'contain'`
+* `pointerEvents`
+  * Support setting the `pointerEvents` values via style rather than a prop.
+  * Retain the React Native specific `box-none` and `box-only` values.
+* `textShadow`
+  * Add native support for CSS text shadows.
+* `transform`
+  * Support using string values to set transforms.
+  * This can be shimmed in user-space with a parser.
+* `verticalAlign`
+  * Make `verticalAlign` the same as `textAlignVertical`.
+  * `verticalAlign === 'middle'` is the same as `textAlignVertical = 'center'`;
+* `userSelect`
+  * Support setting ability to select text via style.
+  * Equivalent to using `selectable` prop on `<Text>`.
+
+```jsx
+<View
+  style={{
+    boxShadow: '1px 1px 1px 1px #eee',
```

Comment:
Wouldn't it be strange to have _just this one_ CSS-like shorthand (i.e. multiple values separated by spaces) in RN? I understand where this comes from (there's no long-form `box-shadow` in CSS as far as I know), but it seems out of space.

I'd rather this read `boxShadow: [1, 1, 1, 1, 'red']`, and then extend this syntax to at least some common places where in CSS you'd use a shorthand, for example:

```js
margin: [4, 10],
padding: [8, '20%', 15],
border: [1, 'solid', 'red'],
```

having shorthands for these common properties would really improve RN StyleSheet ergonomics, and map cleanly to CSS (while not pretending to be CSS)

Reply:
The point is to avoid introducing non-standard syntax that requires transforms on web and reeducation of developers